crates/gpui/src/platform/linux/platform.rs

```diff
@@ -594,6 +597,46 @@ impl Modifiers {
     }
 }
 
+// From https://github.com/wez/wezterm/blob/95581d8697f3749f84ccb1402ac94ea6582b227f/window/src/os/wayland/window.rs#L468
+pub(super) fn read_pipe_with_timeout(mut file: FileDescriptor) -> Result<String> {
+    let mut result = Vec::new();
+
+    file.set_non_blocking(true)?;
+
+    let mut pfd = libc::pollfd {
+        fd: file.as_raw_fd(),
+        events: libc::POLLIN,
+        revents: 0,
+    };
+
+    let mut buf = [0u8; 8192];
+
+    loop {
+        if unsafe { libc::poll(&mut pfd, 1, 1000) == 1 } {
```

Comment:
Can we set this to 0? I'd prefer not to lock up the application for a whole second. If need be, we can keep re-polling it from elsewhere.

Reply:
The timeout is there in case for some reason the read takes a second, but under normal circumstances it should be instant.

Reply:
Yeah, but I'd rather not have the app randomly freeze.

That said, calloop is already an abstraction over epoll, could we register this file descriptor in calloop and have a handler for it there? Then we wouldn't have to worry about blocking and we wouldn't have to add libc and filedescriptor.

Reply:
Update from discord: After poking at it a little bit, we're going to use normal tasks and background threads for this.